W-19639927-MQ-updateSubscriberDoc-dm (#3002)
mulesoft/docs-connectors: anypoint-mq/4.x/modules/ROOT/pages/anypoint-mq-listener.adoc

[IMPORTANT]
To preserve message processing order, set the value of `maxConcurrency` to `1` for flows that consume messages from a FIFO queue.

[NOTE]

> When you configure a polling subscriber with `fetchSize=1` and `maxConcurrency=1` on a FIFO queue with message groups, the subscriber fetches one message per poll and dispatches it to the flow. Before the first message is acknowledged, the next poll can fetch a second message, which waits to be processed until the flow finishes processing the first one. Both messages are considered in-flight at the broker level until they are acknowledged. This configuration preserves message order but can result in two in-flight messages per message group at a time, which is expected behavior.

Reviewer: ```suggestion
When you configure a polling subscriber with `fetchSize=1` and `maxConcurrency=1` on a FIFO queue with message groups, the subscriber fetches one message per poll and dispatches it to the flow. Before the first message is acknowledged, the next poll can fetch a second message, which waits to be processed until the flow finishes processing the first one. Both messages are considered in-flight at the broker level until they're acknowledged. This configuration preserves message order but can result in two in-flight messages per message group at a time, which is expected behavior.
```

[IMPORTANT]
To preserve message processing order, set the value of `maxConcurrency` to `1` for flows that consume messages from a FIFO queue.

[NOTE]

> ====
> When you configure a polling subscriber with `fetchSize=1` and `maxConcurrency=1` on a FIFO queue with message groups, the subscriber fetches one message per poll and dispatches it to the flow. Before the first message is acknowledged, the next poll can fetch a second message, which waits to be processed until the flow finishes processing the first one. Both messages are considered in-flight at the broker level until they are acknowledged. This configuration preserves message order but can result in two in-flight messages per message group at a time, which is expected behavior.
> ====

Author: ```suggestion
When you configure a polling subscriber with `fetchSize=1` and `maxConcurrency=1` on a FIFO queue with message groups, the subscriber fetches one message per poll and dispatches it to the flow. Before the first message is acknowledged, the next poll can fetch a second message, which waits to be processed until the flow finishes processing the first one. Both messages are considered in-flight at the broker level until they are acknowledged. This configuration preserves message order but can result in two in-flight messages per message group at a time, which is expected behavior.
```

[IMPORTANT]
To preserve message processing order, set the value of `maxConcurrency` to `1` for flows that consume messages from a FIFO queue.

[NOTE]
====

> When you use a polling subscriber with `fetchSize=1` and `maxConcurrency=1` on a FIFO queue with message groups:
> 
> * The subscriber fetches one message per poll and dispatches it to the flow.
> * Before the first message is acknowledged, the next poll can fetch a second message, which stays buffered until the flow finishes processing the first one.
> * Both messages are considered in-flight at the broker level until they are acknowledged.
> 
> This configuration preserves message order but can result in two in-flight messages per message group at a time. This is expected behavior.

Author: ```suggestion
When you configure a polling subscriber with `fetchSize=1` and `maxConcurrency=1` on a FIFO queue with message groups, the subscriber fetches one message per poll and dispatches it to the flow. Before the first message is acknowledged, the next poll can fetch a second message, which waits to be processed until the flow finishes processing the first one. Both messages are considered in-flight at the broker level until they are acknowledged. This configuration preserves message order but can result in two in-flight messages per message group at a time, which is expected behavior.
```
